### ISSUE TITLE
Generalise framework-specific preprocessing to include estimator.preprocessing (standardisation)

### ISSUE THREAD

**Issue by beat-buesser** [Trusted-AI/adversarial-robustness-toolbox#625]
The new framework-specific preprocessing defences of ART 1.4 for PyTorch and TensorFlow v2 allow the implementation of preprocessing defences and the backpropagation of gradients through them using the framework's methods and automated gradient calculation. 

The estimator of ART 1.4 support standardising the input with `(x - mean) / std` which is sufficient for some models for image classification but does not cover all standardisation methods of inputs.

This issue should build on the framework-specific preprocessing defences to provide support for general preprocessing (standardisation) functions.

These new methods for standardisation should become part of a new module `art.preprocessing`. This new module could also provide support for preprocessing discrete input data.
